util/git/creds_test.go

```diff
@@ -43,21 +43,18 @@ func (s *memoryCredsStore) Remove(id string) {
 	delete(s.creds, id)
 }
 
+func (s *memoryCredsStore) Environ(id string) []string {
+	return nil
+}
+
 func TestHTTPSCreds_Environ_no_cert_cleanup(t *testing.T) {
 	store := &memoryCredsStore{creds: make(map[string]cred)}
 	creds := NewHTTPSCreds("", "", "", "", true, "", store, false)
-	closer, env, err := creds.Environ()
+	closer, _, err := creds.Environ()
 	require.NoError(t, err)
-	var nonce string
-	for _, envVar := range env {
-		if strings.HasPrefix(envVar, ASKPASS_NONCE_ENV) {
-			nonce = envVar[len(ASKPASS_NONCE_ENV)+1:]
-			break
-		}
-	}
-	assert.Contains(t, store.creds, nonce)
```

Comment:
I am not sure but was this test case moved/validated elsewhere? If not, maybe good to have a similar validation in the new package?

Reply:
Adding `ASKPASS_NONCE_ENV` to the CredsStore is store-specific (so, the askpass implementation of CredsStore adds that env var, but the in-memory implementation does not).

Since it's now `store` adding that env var instead of `creds`, and since `store` is now `memoryCredsStore` instead of the askpass creds store, the askpass-specific env var isn't added.

This test basically asserts that the `creds` environ closer removes the relevant credential from the creds store. Checking length instead of specific contents just makes the test store-agnostic.

Removal of the askpass-specific element is covered in the askpass `Remove` unit test.

Reply:
(Sorry, that's long-winded, but good to be extra clear since it's a security mechanism being tested. :-))